File: android-app/app/src/main/java/com/example/myapplication/MainActivity.java
package com.example.myapplication;

import android.os.Bundle;
import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;
import androidx.fragment.app.Fragment;
import com.google.android.material.bottomnavigation.BottomNavigationView;

public class MainActivity extends AppCompatActivity {
    private BottomNavigationView bottomNavigationView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        EdgeToEdge.enable(this);

        ViewCompat.setOnApplyWindowInsetsListener(findViewById(android.R.id.content), (v, insets) -> {
            int systemBars = WindowInsetsCompat.Type.systemBars();
            v.setPadding(insets.getInsets(systemBars).left,
                    insets.getInsets(systemBars).top,
                    insets.getInsets(systemBars).right,
                    insets.getInsets(systemBars).bottom);
            return insets;
        });

        bottomNavigationView = findViewById(R.id.bottomNavigationView);
        bottomNavigationView.setOnItemSelectedListener(menuItem -> {
            Fragment selectedFragment = null;

            int itemId = menuItem.getItemId();

            if (itemId == R.id.nav_home) {
                selectedFragment = new HomeFragment();
            } else if (itemId == R.id.nav_categories) {
                selectedFragment = new CategoryFragment();
            } else if (itemId == R.id.nav_news) {
                selectedFragment = new NewsFragment();
            } else if (itemId == R.id.nav_bookmark) {
                selectedFragment = new BookmarkFragment();
            } else if (itemId == R.id.nav_settings) {
                selectedFragment = new SettingsFragment();
            }

            if (selectedFragment != null) {
                replaceFragment(selectedFragment);
            }
            return true;
        });

        if (savedInstanceState == null) {
            bottomNavigationView.setSelectedItemId(R.id.nav_home);
        }
    }

    private void replaceFragment(Fragment fragment) {
        Fragment currentFragment = getSupportFragmentManager().findFragmentById(R.id.frame_container);
        if (currentFragment != null && currentFragment.getClass().equals(fragment.getClass())) {
            return;
        }

        getSupportFragmentManager().beginTransaction()
                .replace(R.id.frame_container, fragment)
                .commit();
    }
}

File: app/src/main/java/com/example/newsapp/MainActivity.kt
package com.example.newsapp

import Category
import News
import android.os.Bundle
import android.view.View
import android.widget.GridView
import android.widget.ImageView
import android.widget.ListView
import android.widget.TextView
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import androidx.viewpager2.widget.ViewPager2
import com.example.newsapp.Adapter.CategoryAdapter
import com.example.newsapp.Adapter.ImageSliderAdapter
import com.example.newsapp.Adapter.NewsAdapter

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.help)
        this.enableEdgeToEdge()

        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.helpCenter)) { v: View, insets: WindowInsetsCompat ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

//        HelpCenter
        val answerText = findViewById<TextView>(R.id.answerText)
        val toggleButton = findViewById<ImageView>(R.id.toggleButton)

        toggleButton.setOnClickListener {
            if (answerText.visibility == View.VISIBLE) {
                answerText.visibility = View.GONE
                toggleButton.setImageResource(R.drawable.down_arrow)
            } else {
                answerText.visibility = View.VISIBLE
            }
        }

//        Category
//        val categories = listOf(
//            Category("Technology", R.drawable.news1),
//            Category("Sports", R.drawable.news2),
//            Category("Health", R.drawable.news3),
//        )
//        val gridView = findViewById<GridView>(R.id.categoriesGridView)
//        gridView.adapter = CategoryAdapter(this, categories)


//        MainScreen
//        val viewPager = findViewById<ViewPager2>(R.id.viewPager)
//        val images = listOf(
//            R.drawable.news1,
//            R.drawable.news2
//        )
//        val titles = listOf(
//            "Let’s settle the debate: IOS vs Androidz Consumer verdict",
//            "Let’s settle the debate: IOS vs Androidz Consumer verdict"
//        )
//        val categories = listOf(
//            "Sport",
//            "Technology"
//        )
//        val adapterSlider = ImageSliderAdapter(images, titles, categories)
//        viewPager.adapter = adapterSlider
//
//        val newsListView = findViewById<ListView>(R.id.newsListView)
//        val newsList = listOf(
//            News("Technology", "IOS vs Android Consumer verdict", "Sun 9 March, 2025", R.drawable.news3),
//            News("Sports", "Latest match highlights: Epic showdown", "Mon 10 March, 2025", R.drawable.news3),
//            News("Health", "The benefits of a balanced diet", "Tue 11 March, 2025", R.drawable.news3),
//            News("Law", "New legal regulations coming soon", "Wed 12 March, 2025", R.drawable.news3)
//        )
//        val adapterNews = NewsAdapter(this, newsList)
//        newsListView.adapter = adapterNews
    }
}
